File: prob3_bronze_season26contest1/java/TestRunner.java
import java.io.IOException;
import java.io.PrintWriter;
import java.net.URISyntaxException;
import java.nio.file.DirectoryStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;
import java.util.Locale;
import java.util.Scanner;

public class TestRunner {
    private static List<String> runFile(Path file) throws IOException {
        List<String> outputs = new ArrayList<>();
        try (Scanner sc = new Scanner(Files.newBufferedReader(file))) {
            if (!sc.hasNextInt()) {
                return outputs;
            }
            int n = sc.nextInt();
            int k = sc.nextInt();
            int q = sc.nextInt();
            List<Solution.Update> updates = new ArrayList<>(q);
            for (int i = 0; i < q; i++) {
                int r = sc.nextInt() - 1;
                int c = sc.nextInt() - 1;
                long v = sc.nextLong();
                updates.add(new Solution.Update(r, c, v));
            }
            for (long value : Solution.solveAll(n, k, updates)) {
                outputs.add(Long.toString(value));
            }
        }
        return outputs;
    }

    private static List<String> readExpected(Path file) throws IOException {
        List<String> outputs = new ArrayList<>();
        try (Scanner sc = new Scanner(Files.newBufferedReader(file))) {
            while (sc.hasNext()) {
                outputs.add(sc.next());
            }
        }
        return outputs;
    }

    private static Path getBaseDir() throws URISyntaxException {
        Path location = Paths.get(TestRunner.class.getProtectionDomain()
                .getCodeSource().getLocation().toURI());
        if (Files.isRegularFile(location)) {
            return location.getParent();
        }
        return location;
    }

    private static Path findTestDir() throws URISyntaxException {
        Path cwd = Paths.get(System.getProperty("user.dir"));
        Path testDir = cwd.resolve("testData");
        if (Files.isDirectory(testDir)) {
            return testDir;
        }
        Path baseDir = getBaseDir();
        return baseDir.resolve("..").resolve("testData").normalize();
    }

    public static void main(String[] args) throws Exception {
        Path testDir = findTestDir();
        if (!Files.isDirectory(testDir)) {
            System.out.println("testData folder not found: " + testDir.toAbsolutePath());
            return;
        }

        List<Path> files = new ArrayList<>();
        try (DirectoryStream<Path> stream = Files.newDirectoryStream(testDir, "*.in")) {
            for (Path p : stream) {
                files.add(p);
            }
        }
        Collections.sort(files, new Comparator<Path>() {
            @Override
            public int compare(Path a, Path b) {
                String aName = a.getFileName().toString();
                String bName = b.getFileName().toString();
                if (aName.length() != bName.length()) {
                    return Integer.compare(aName.length(), bName.length());
                }
                return aName.compareTo(bName);
            }
        });

        PrintWriter out = new PrintWriter(System.out);
        long totalStart = System.nanoTime();
        for (Path p : files) {
            Path outPath = testDir.resolve(p.getFileName().toString().replace(".in", ".out"));
            long caseStart = System.nanoTime();
            List<String> outputs = runFile(p);
            boolean hasExpected = Files.exists(outPath);
            List<String> expected = hasExpected ? readExpected(outPath) : new ArrayList<>();

            String status;
            if (!hasExpected) {
                status = "MISSING .out";
            } else if (outputs.equals(expected)) {
                status = "PASS";
            } else {
                status = "FAIL";
            }

            double elapsedMs = (System.nanoTime() - caseStart) / 1_000_000.0;
            out.println(p.getFileName() + ": " + status + String.format(Locale.US, " (%.2f ms)", elapsedMs));
        }
        double totalMs = (System.nanoTime() - totalStart) / 1_000_000.0;
        out.println(String.format(Locale.US, "Total: %.2f ms", totalMs));
        out.flush();
    }
}

File: prob3_bronze_season26contest1/python/test_runner.py
import os
import time

import solution


def run_file(path):
    with open(path, "r", encoding="utf-8") as f:
        data = f.read().strip().split()
    if not data:
        return []
    n = int(data[0])
    k = int(data[1])
    q = int(data[2])
    idx = 3
    updates = []
    for _ in range(q):
        r = int(data[idx]) - 1
        c = int(data[idx + 1]) - 1
        v = int(data[idx + 2])
        idx += 3
        updates.append((r, c, v))
    return [str(x) for x in solution.solve_all(n, k, updates)]


def read_expected(path):
    with open(path, "r", encoding="utf-8") as f:
        content = f.read().strip()
    if not content:
        return []
    return content.split()


def main():
    base_dir = os.path.dirname(os.path.abspath(__file__))
    test_dir = os.path.abspath(os.path.join(base_dir, "..", "testData"))
    if not os.path.isdir(test_dir):
        print("testData folder not found:", test_dir)
        return

    files = [f for f in os.listdir(test_dir) if f.endswith(".in")]
    files.sort(key=lambda x: (len(x), x))

    total_start = time.perf_counter()
    for name in files:
        in_path = os.path.join(test_dir, name)
        out_name = name.replace(".in", ".out")
        out_path = os.path.join(test_dir, out_name)

        case_start = time.perf_counter()
        outputs = run_file(in_path)
        expected = read_expected(out_path) if os.path.exists(out_path) else None

        if expected is None:
            status = "MISSING .out"
        elif outputs == expected:
            status = "PASS"
        else:
            status = "FAIL"

        elapsed_ms = (time.perf_counter() - case_start) * 1000.0
        print(f"{name}: {status} ({elapsed_ms:.2f} ms)")

    total_ms = (time.perf_counter() - total_start) * 1000.0
    print(f"Total: {total_ms:.2f} ms")


if __name__ == "__main__":
    main()
